src/pytest_inmanta_lsm/lsm_project.py

```diff
@@ -671,6 +672,53 @@ def get_service_entity(self, service_entity_name: str) -> inmanta_lsm.model.Serv
 
         return self.service_entities[service_entity_name]
 
+    def transfer_to_next_state(
+        self,
+        service: inmanta_lsm.model.ServiceInstance,
+        transfer: inmanta_lsm.model.LifecycleTransfer,
+        is_error_transition: bool = False,
+    ) -> None:
+        """
+        Apply this state to our service, if it is different from the current
+        state, also increment the version and the desired state version (if the exported model is changing)
+
+        :param service: The current service
+        :param transfer: The current transfer
+        :param is_error_transition: Is it an error transition?
+        """
+        if is_error_transition:
+            if transfer is None:
+                raise RuntimeError(f"Error transition is not defined for {str(transfer)}!")
+
+            next_state = transfer.error
+        else:
+            next_state = transfer.target
+
+        if service.state == next_state:
+            return
+
+        # In case of an `AttributeError`, we don't need to do anything: we are dealing with an old orchestrator
+        is_preserving_same_desired_state = False
+        with suppress(AttributeError):
+            if is_error_transition:
+                is_preserving_same_desired_state = transfer.error_same_desired_state
+            else:
+                is_preserving_same_desired_state = transfer.target_same_desired_state
+
+            if not is_preserving_same_desired_state:
+                service.desired_state_version += 1
```

Comment:
```suggestion
        if not is_preserving_same_desired_state:
            service.desired_state_version += 1
```
We also have to do it for the default, right?